mods/default/tools.lua

```diff
@@ -410,9 +410,10 @@ minetest.register_tool("default:skeleton_key", {
 				-- finish and return the new key
 				itemstack:take_item()
 				itemstack:add_item("default:key")
-				itemstack:set_metadata(minetest.write_json({
-					secret = secret
-				}))
+				local meta = itemstack:get_meta()
+				meta:set_string("secret", secret)
+				meta:set_string("description", minetest.registered_tools["default:key"].description
+					.." (owned by "..placer:get_player_name()..")")
```

Comment:
Done. Resulting description would look something like this: "Key to a Steel Door (owned by singleplayer)"